src/sagemaker_pytorch_serving_container/torchserve.py

```diff
@@ -0,0 +1,215 @@
+        DEFAULT_TS_MODEL_DIRECTORY,
+        "--extra-files",
+        os.path.join(environment.model_dir, environment.Environment().module_name + ".py"),
+        "--version",
+        "1",
+    ]
+
+    logger.info(model_archiver_cmd)
+    subprocess.check_call(model_archiver_cmd)
+
+    _set_python_path()
+
+
+def _set_python_path():
+    # Torchserve handles code execution by appending the export path, provided
+    # to the model archiver, to the PYTHONPATH env var.
+    # The code_dir has to be added to the PYTHONPATH otherwise the
+    # user provided module can not be imported properly.
+    code_dir_path = "{}:".format(environment.code_dir)
+
+    if PYTHON_PATH_ENV in os.environ:
+        os.environ[PYTHON_PATH_ENV] = code_dir_path + os.environ[PYTHON_PATH_ENV]
+    else:
+        os.environ[PYTHON_PATH_ENV] = code_dir_path
+
+
+def _create_torchserve_config_file():
+    configuration_properties = _generate_ts_config_properties()
+
+    utils.write_file(TS_CONFIG_FILE, configuration_properties)
+
+
+def _generate_ts_config_properties():
+    env = environment.Environment()
+
+    user_defined_configuration = {
+        "default_response_timeout": env.model_server_timeout,
+        "default_workers_per_model": env.model_server_workers,
+        "inference_address": "http://0.0.0.0:{}".format(env.inference_http_port),
+        "management_address": "http://0.0.0.0:{}".format(env.management_http_port),
+    }
+
+    custom_configuration = str()
+
+    for key in user_defined_configuration:
+        value = user_defined_configuration.get(key)
+        if value:
+            custom_configuration += "{}={}\n".format(key, value)
+
+    if ENABLE_MULTI_MODEL:
+        default_configuration = utils.read_file(MME_TS_CONFIG_FILE)
+    else:
+        default_configuration = utils.read_file(DEFAULT_TS_CONFIG_FILE)
+
+    return default_configuration + custom_configuration
+
+
+def _add_sigterm_handler(ts_process):
+    def _terminate(signo, frame):  # pylint: disable=unused-argument
+        try:
+            os.kill(ts_process.pid, signal.SIGTERM)
+        except OSError:
+            pass
+
+    signal.signal(signal.SIGTERM, _terminate)
+
+
+def _install_requirements():
+    logger.info("installing packages from requirements.txt...")
+    pip_install_cmd = [sys.executable, "-m", "pip", "install", "-r", REQUIREMENTS_PATH]
+
+    try:
+        subprocess.check_call(pip_install_cmd)
+    except subprocess.CalledProcessError:
+        logger.error("failed to install required packages, exiting")
+        raise ValueError("failed to install required packages")
+
+
+# retry for 10 seconds
+@retry(stop_max_delay=10 * 1000)
+def _retrieve_ts_server_process():
+    ts_server_processes = list()
+
+    for process in psutil.process_iter():
+        if TS_NAMESPACE in process.cmdline():
+            ts_server_processes.append(process)
+
+    if not ts_server_processes:
+        raise Exception("ts model server was unsuccessfully started")
```

Comment:
let's write out what "TS" stands for since these are more visible exception messages. same in l. 213

```suggestion
        raise Exception("Torchserve model server was unsuccessfully started")
```